core/src/coproduct.rs

```diff
@@ -547,6 +552,75 @@ impl<Choices> CoproductSubsetter<CNil, HNil> for Choices {
     { Err(self) }
 }
 
+/// Trait for upcasting a coproduct into another that can hold its variants.
```

Comment:
Hmm... is "upcasting" the right word here?

Reply:
It is a bit misleading, I agree.  (my original goal when first designing it was to make something that does upcasting; but the final implementation is capable of doing other things as well)

What I initially wrote here was "Trait for infallible conversions between coproducts", though I was afraid somebody might ask why not use `From/Into` for this.  (to which my reply would be: we only get one chance to write a generic from/into, so we have to spend it carefully)

Reply:
Can we just say the following and let the examples explain the semantics?
> Trait for **embedding** a coproduct into another that can hold its variants.

Reply:
I was thinking of how it would look on e.g. the module page (where that first line is the only information a user is given). How will a user know that "embedding" is what they want?

That said, this concern won't matter once inherent method wrappers are added (in which case the first line doesn't show up out of context anywhere, and so "embedding" would work just fine).

Reply:
How about just

> Trait for **converting** a coproduct into another that can hold its variants.

Reply:
SGTM 👍